### ISSUE TITLE
Form Submission Cookie Not Automatically Setting "Form" Key

### ISSUE THREAD

**Issue by fpcorso** [PopupMaker/Popup-Maker#866]
## Describe the bug
When generating any trigger and automatically adding a "Form Submission" cookie, the cookie gets created but the `form` key does not get set. This causes the cookie to not be set. The issue is resolved by editing the cookie and either making no changes or setting the form to any form. Once updated, the `form` key will be valid.

## Site information
Popup Maker version: 1.11.1
WordPress version: 5.4

## Expected behavior
Should automatically set the `form` key.

## Current behavior
`form` key is not set.

## Steps to reproduce
1. Create popup
2. Create any trigger, leaving "Prevent popup from showing to visitor again using a cookie?" checked and "Form Submission" as the cookie.
3. Save popup
4. Review values to see no `form` key is present: 
![image](https://user-images.githubusercontent.com/9730040/89306726-427c9a00-d63e-11ea-90b5-ad2615715e5a.png)
5. Edit the cookie but make no changes. Click "Update" on cookie modal. Save popup.
6. Review values to see the new `form` key:
![image](https://user-images.githubusercontent.com/9730040/89306840-650eb300-d63e-11ea-8c6b-5304073f29b3.png)

## Errors
No JavaScript errors are shown during cookie creation or editing.

**Comment by fpcorso** [comment]
@danieliser I think this may affect any cookie that has any extra settings added. In triggers.js, we check to see if the checkbox is checked and set the time, path, and event type here: https://github.com/PopupMaker/Popup-Maker/blob/master/assets/js/src/admin/popup-editor/plugins/triggers.js#L431

Which we then pass to `insertCookie` over in cookies.js which only extends to make sure there is an event type and a name: https://github.com/PopupMaker/Popup-Maker/blob/master/assets/js/src/admin/popup-editor/plugins/cookies.js#L97

That then gets added to the editor through the `row.add` method in the same file: https://github.com/PopupMaker/Popup-Maker/blob/master/assets/js/src/admin/popup-editor/plugins/cookies.js#L201


Unless I am mistaken, I don't think Popup Maker ever adds in any default settings for any extra setting field that is added to a cookie, such as `form` in this case. Is there anything I might have missed here?

**Comment by danieliser** [comment]
@fpcorso - It is one of the older systems still in place, very well possible it has no default value merge.

Try it and see what happens is all we can do.

**Comment by fpcorso** [comment]
@danieliser After a bit more debugging, I realized there are other settings that haven't been included too due to this. For example, the session-wide setting was never set for any cookie if you never manually edited the cookie. Of course, our checks also checked if it was set so it didn't affect anything unless they went in and turned it on so it wasn't an issue anyone would notice.

So, I set up a getCookieDefault method which flattens the default fields into the settings object and then used that for the `extend` instead of our hardcoded one. After a bit of testing, looks like this solves the issue and has no side-effects with any other cookie creation or editing.
